cmd/admin-prometheus-generate.go

```diff
@@ -166,38 +159,26 @@ func generatePrometheusConfig(ctx *cli.Context) error {
 		return nil
 	}
 
-	u, err := url.Parse(hostConfig.URL)
-	if err != nil {
-		return err
+	u, e := url.Parse(hostConfig.URL)
+	if e != nil {
+		return e
 	}
 
-	jwt := jwtgo.NewWithClaims(jwtgo.SigningMethodHS512, jwtgo.StandardClaims{
-		ExpiresAt: UTCNow().Add(defaultPrometheusJWTExpiry).Unix(),
-		Subject:   hostConfig.AccessKey,
-		Issuer:    "prometheus",
-	})
+	if !ctx.Bool("public") {
+		jwt := jwtgo.NewWithClaims(jwtgo.SigningMethodHS512, jwtgo.StandardClaims{
+			ExpiresAt: UTCNow().Add(defaultPrometheusJWTExpiry).Unix(),
+			Subject:   hostConfig.AccessKey,
+			Issuer:    "prometheus",
+		})
 
-	token, err := jwt.SignedString([]byte(hostConfig.SecretKey))
-	if err != nil {
-		return err
-	}
-	client, cerr := newAdminClient(alias)
-	fatalIf(cerr, "Unable to initialize admin connection.")
+		token, e := jwt.SignedString([]byte(hostConfig.SecretKey))
+		if e != nil {
+			return e
+		}
 
-	info, e := client.ServerInfo(globalContext)
-	if e != nil {
-		fatalIf(probe.NewError(e), "Failed to get server info.")
+		// Setting the values
+		defaultConfig.ScrapeConfigs[0].BearerToken = token
 	}
-	if info.Servers[0].Version < "2021-01-30T00-20-58Z" {
```

Comment:
2021-01-30T00-20-58Z is considered very recent for systems in production :) I think it is still so soon to remove legacy code.

Reply:
yeah but this is not useful @vadmeste we can simply document this properly and whoever is 01-30 can use older `mc`. We don't have to support older servers for long periods.